emu_sv/state_vector.py

```diff
@@ -140,7 +159,9 @@ def _index_to_bitstring(self, index: int) -> str:
         """
         Convert an integer index into its corresponding bitstring representation.
         """
-        nqubits = int(math.log2(self.vector.reshape(-1).shape[0]))
+        nqubits = self.n_qudits
+        msg = f"index {index} can not exceed Hilbert space size d**{nqubits}"
+        assert index < 2**nqubits, msg
```

Comment:
I dislike these asserts in private code. We are the only users, and we have unit tests. The asserts make everything slow, and they are unlikely to cause any errors that the tests will not.